COMMERCE-1309 Multi-Site Catalog

ethanbustad opened the pull request:
https://issues.liferay.com/browse/COMMERCE-1310
https://issues.liferay.com/browse/COMMERCE-1312
https://issues.liferay.com/browse/COMMERCE-1313
https://issues.liferay.com/browse/COMMERCE-1314
https://issues.liferay.com/browse/COMMERCE-1316
https://issues.liferay.com/browse/COMMERCE-1317
https://issues.liferay.com/browse/COMMERCE-1318

Work in progress.

alecsloan committed:
COMMERCE-149 Fix Terms of Use panel

Commit:
COMMERCE-X deployment instructions

Commit:
COMMERCE-1414 Source Formatting

Commit:
COMMERCE-1414 Add blogs support

Commit:
COMMERCE-1414 Add discounts

Commit:
COMMERCE-1414 Add forms

Commit:
COMMERCE-1414 Add documents support

Commit:
COMMERCE-1414 Add web content support

Commit:
COMMERCE-1414 Add KB Article support

Commit:
COMMERCE-1414 Misc changes to add portlets to page etc

Commit:
COMMERCE-1414 Discount category

alecsloan committed:
COMMERCE-1246 Make zip a required model field

alessiorendina committed:
COMMERCE-1234 Use DTO mapper for each resource

alessiorendina committed:
COMMERCE-1234 Refactor base helper into expando utility class

alessiorendina committed:
COMMERCE-1234 Added date config utility class

alessiorendina committed:
COMMERCE-1234 Use utility classes instead of ds components for resources

alessiorendina committed:
COMMERCE-1234 Refactor yaml

alessiorendina committed:
COMMERCE-1234 Added new method to commerce product definition link local and remote service implementation

alessiorendina committed:
COMMERCE-1234 Refactor catalog resource implementations

alessiorendina committed:
COMMERCE-1234 Regen rest module

alessiorendina committed:
COMMERCE-1234 Regen services

alecsloan committed:
COMMERCE-935 Only show cancel button if backURL exists

Commit:
COMMERCE-1392 Add default order renderer view for orders submitted with
a workflow

alessiorendina committed:
COMMERCE-704 Refactor externalReferenceCode from CPDefinition to CProduct

alessiorendina committed:
COMMERCE-704 Refactor commerce product definition local and remote service implementation

alessiorendina committed:
COMMERCE-704 Refactor commerce product local service implementation

alessiorendina committed:
COMMERCE-704 Added new exceptions

alessiorendina committed:
COMMERCE-704 Use project dependency

alessiorendina committed:
COMMERCE-704 Validate external reference code

alessiorendina committed:
COMMERCE-704 Use CProduct external reference code instead

alessiorendina committed:
COMMERCE-704 Increment schema version

alessiorendina committed:
COMMERCE-704 Fixed commerce product upgrade process

alessiorendina committed:
COMMERCE-704 Added new upgrade process

alessiorendina committed:
COMMERCE-704 Regen services

marco-leo committed:
COMMERCE-1414 Fix merge

marco-leo committed:
COMMERCE-1414 SF

alecsloan committed:
COMMERCE-1196 Set checkoutURL even when order is null

alecsloan committed:
EMP-25 Breccia Images should open in modal

gcmznt committed:
COMMERCE-686 fix product gallery ratio

alessiorendina committed:
COMMERCE-1234 Removed no more used helpers

alessiorendina committed:
COMMERCE-1234 Refacto commerce product definition rel local and remote service implementation

alessiorendina committed:
COMMERCE-1234 Refactor commerce product definition link local and remote service implementation

alecsloan committed:
COMMERCE-1299 Use CommerceAccountId instead of UserId for addresses

alessiorendina committed:
COMMERCE-704 Use project dependencies

alessiorendina committed:
COMMERCE-1234 Semantic versioning (gradlew baseline)

Commit:
EMP-26 added new paypal payment item for taxes

Commit:
EMP-26 new language property for paypal taxes

Commit:
EMP-26 build lang

alecsloan committed:
COMMERCE-1298 Add error handling for duplicate CPAttachmentFileEntry

alecsloan committed:
COMMERCE-1298 gradle buildLang

Commit:
COMMERCE-1303 an order is free shipping if all items are free shipping

Commit:
COMMERCE-1303 if order became free shipping, remove shipping method and costs

alecsloan committed:
COMMERCE-1302 Order and Order Item's custom fields section should have their own tab for isVisible to function

alecsloan committed:
COMMERCE-1298 gradle baseline

alecsloan committed:
COMMERCE-1310 Implement CommerceCatalog schema

alecsloan committed:
COMMERCE-1310 Implement basic add/update/delete catalog methods

alecsloan committed:
COMMERCE-1310 gradle buildService

Commit:
COMMERCE-1310 Suppress SF persistence check for GroupPersistence call

Portal provides no other way to access this finder than to call
GroupPersistence directly :/

alecsloan committed:
COMMERCE-1316 Implement CommerceCatalogUpgradeProcess

alecsloan committed:
COMMERCE-1316 Implement Catalogs UI in 'Commerce' section in Control Panel for now

Commit:
COMMERCE-1312 Create interfaces for creating and using catalog indexes

Commit:
COMMERCE-1312 Create implementations for creating and using catalog indexes

Commit:
COMMERCE-1312 Create catalog indexes on portal startup

Commit:
COMMERCE-1312 Use newer search API version to get UpdateDocRequest.setType

alecsloan committed:
COMMERCE-1316 gradle buildLang

alecsloan committed:
COMMERCE-1316 Implement basic remote service without permission check for action command usage

alecsloan committed:
COMMERCE-1316 gradle buildService

Commit:
COMMERCE-1316 Remove unused code / simplify

Commit:
COMMERCE-1316 Remove parentCommerceCatalogId

Commit:
COMMERCE-1310 Remove parentCommerceCatalogId: use channels

alecsloan committed:
COMMERCE-1316 Implement CommerceCatalogScopeHelperRegistry so we can search the index in services

Commit:
COMMERCE-1316 These permissions should be restrictive by default

alecsloan committed:
COMMERCE-1316 Use the CatalogScopeHelper to search all catalogs and list them in the portlet

Commit:
COMMERCE-1312 Add method to remove documents from the index

Commit:
COMMERCE-1314 Add tables to service.xml for Channels

Commit:
COMMERCE-1314 Regenerate (gw buildService)

Commit:
COMMERCE-1314 Fix generated files

alecsloan committed:
COMMERCE-1313 Pass catalogId in toolbar tabs

alecsloan committed:
COMMERCE-1313 Hide Catalog from the site scope

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

alessiorendina committed:
COMMERCE-1234 Refactor commerce product definition local and remote service implementation

alessiorendina committed:
COMMERCE-1234 Added dto converter

alessiorendina committed:
COMMERCE-1234 Added dto converter registry

alessiorendina committed:
COMMERCE-1234 Replaced dto mappers to dto converters

alessiorendina committed:
COMMERCE-1234 Use project dependency instead

alessiorendina committed:
COMMERCE-1234 Avoid methods that are called only once

alessiorendina committed:
COMMERCE-1234 Refactor catalog resource implementations to use dto converter instead

alessiorendina committed:
COMMERCE-1234 Auto SF

Commit:
LPS-94365 Use latest arquillian version to speed up tests

Commit:
LPS-94365 Provide missing dependency (not sure how this ever worked)

alecsloan committed:
COMMERCE-1318 Add getters to remote service

alecsloan committed:
COMMERCE-1318 Implement Catalog constants

alecsloan committed:
COMMERCE-1318 Add constant for Account role type

alecsloan committed:
COMMERCE-1318 Implement CatalogRoleHelper and Catalog Role health status

alecsloan committed:
COMMERCE-1318 Refactor userRoleModal taglib so its usable for Catalogs

alecsloan committed:
COMMERCE-1318 Minor navigationItem fixes

alecsloan committed:
COMMERCE-1318 Implement Catalog Details/Users sections

alecsloan committed:
COMMERCE-1318 Remove Catalog section from site context

alecsloan committed:
COMMERCE-1318 Localize catalog names in list

alecsloan committed:
COMMERCE-1318 gradle buildService

alecsloan committed:
COMMERCE-1318 gradle buildLang

alecsloan committed:
COMMERCE-1318 gradle baseline

NaderJafari committed:
EMP-27 Apply Order by DESC in finder custom sql to show latest orders on top

alecsloan committed:
COMMERCE-1318 Add Commerce Panel Category Key

gianmarcobrunialti committed:
COMMERCE-686 Add to cart - all buttons

gianmarcobrunialti committed:
COMMERCE-686 - Add to cart buttons - SF

gianmarcobrunialti committed:
COMMERCE-686 - Add to cart buttons

jbalsas committed:
COMMERCE-XXX Prefixes table cell variants with commerceTableCell to avoid name collision with clay default naming schema

alecsloan committed:
COMMERCE-1312 Implement CommerceCatalogIndexer and use for searching

alecsloan committed:
COMMERCE-1312 gradle buildService

alecsloan committed:
COMMERCE-1312 gradle baseline

NaderJafari committed:
EMP-28 Use commerce account id instead of user id

Commit:
COMMERCE-1359 user and account now styled for bootstrap

alecsloan committed:
COMMERCE-1362 Since we need assetLinkEntryIds lets use those instead of assetLinkIds for updating links

Commit:
COMMERCE-1356 COMMERCE-1363 Injected services shouldn't be static

alecsloan committed:
COMMERCE-1314 Add a name field to the Channel entity

Commit:
COMMERCE-1355 Order of results depends on db, so sort before using

NaderJafari committed:
EMP-29 Fixed edit order fieldset

Commit:
COMMERCE-686 Auto SF

Commit:
COMMERCE-686 Manual SF improvement... SF doesn't seem to do much here

alecsloan committed:
COMMERCE-1299 Use accountHelper in addressDisplayContext instead of the serviceUtil

Commit:
COMMERCE-149 Fix compileJSP error

alecsloan committed:
COMMERCE-1321 Add methods to Channel services

alecsloan committed:
COMMERCE-1321 Add Channel upgrade process to create DB table

alecsloan committed:
COMMERCE-1321 gradle buildService

alecsloan committed:
COMMERCE-1321 Semantic versioning (gradle baseline)

alecsloan committed:
COMMERCE-1321 gradle buildLang

alecsloan committed:
COMMERCE-1321 Implement channel types including 2 OOTB and extension point registry

Commit:
COMMERCE-686 fixes on add to cart button

Commit:
COMMERCE-686 minicart restyled

Commit:
COMMERCE-686 Auto SF

alecsloan committed:
COMMERCE-1284 We need language keys in portlet modules to translate them in the Commerce panel

alecsloan committed:
COMMERCE-1284 gradle buildLang

Commit:
COMMERCE-1318 Adjustments for Account Roles and CPRules

alecsloan committed:
COMMERCE-1321 Implement CommerceChannelTypeJSPContributor and Registry extension points

alecsloan committed:
COMMERCE-1321 Implement Organization and Site JSP contributors

Commit:
COMMERCE-1359 edit user and edit account now work just with clay css

riccardo-ferrari committed:
COMMERCE-1374 commerce-product-service Preserve parameters when generating URLs

liferay-translation committed:
LPS-77699 Update Translations

alessiorendina committed:
COMMERCE-1236 Use project dependency instead

alessiorendina committed:
COMMERCE-1236 Refactor commerce account organization and user rel remote service implementations

alessiorendina committed:
COMMERCE-1236 Moved dto converter interfaces into headless commerce core

alessiorendina committed:
COMMERCE-1236 Added exception mappers

alessiorendina committed:
COMMERCE-1236 Added new dto converters for accounts

alessiorendina committed:
COMMERCE-1236 Added account resurce implementations

alessiorendina committed:
COMMERCE-1236 Added headless commerce admin account api and implementation module skeletons

alessiorendina committed:
COMMERCE-1236 Refactor catalog dto converters

alessiorendina committed:
COMMERCE-1236 Regen services

alessiorendina committed:
COMMERCE-1236 Regen rest module

alessiorendina committed:
COMMERCE-1236 Semantic versioning (gradlew baseline)

Commit:
COMMERCE-1318 Add catalog web to the portal build profile

alecsloan committed:
COMMERCE-1321 Implement simple site item selector to select multiple sites from a list

Commit:
COMMERCE-1321 Use className/classPK for CPRules, not scope

alecsloan committed:
COMMERCE-1321 Implement Channel navigation and blank filters page

Commit:
COMMERCE-1321 Add service to link rule to channel

alecsloan committed:
COMMERCE-1318 fix language key

liferay-translation committed:
LPS-77699 Update Translations

alecsloan committed:
COMMERCE-1381 Fix jsp's path and render command's name

alecsloan committed:
COMMERCE-1318 Catalog's Order Manager needs to be distinct as to not clash with Account's

alecsloan committed:
COMMERCE-1319 Implement wrapper for CPCatalogRuleDisplaycontext

alecsloan committed:
COMMERCE-1319 Make use of the new display context wrapper and finalize filters.jsp

alecsloan committed:
COMMERCE-1319 Add a scope to CPRuleTypes

liferay-translation committed:
LPS-77699 Update Translations

alecsloan committed:
COMMERCE-1319 Make name and type required Channel fields

Commit:
COMMERCE-1321 Implement Channels UI, sidebar panel, and display context

Commit:
COMMERCE-1321 Move Channel UI content into its own module

alecsloan committed:
COMMERCE-1319 Replace nameMap with name in channel services since it's no longer localized

alecsloan committed:
COMMERCE-1321 remove unused parameter from method and add search count for catalog remote service

alecsloan committed:
COMMERCE-1321 Fix site name not being localized and block xss

alecsloan committed:
COMMERCE-1321 Implement Catalog Channel type

Commit:
COMMERCE-1340 minium gallery on breccia

liferay-translation committed:
LPS-77699 Update Translations

alessiorendina committed:
COMMERCE-1236 Added source formatter suppressions

alessiorendina committed:
COMMERCE-1234 SF

Commit:
COMMERCE-1390 Add language key values to CommerceOrderTable

Commit:
COMMERCE-1392 Switch to use a language key

alecsloan committed:
COMMERCE-1318 Implement Catalog Details/Users sections SQUASH (This makes users a search container)

alecsloan committed:
COMMERCE-1393 Pass checked definitions in url on pagination change and remove duplicate pagination

alecsloan committed:
COMMERCE-1287 Prevent XSS on some sku outputs

alecsloan committed:
COMMERCE-1292 Prevent xss on price

liferay-translation committed:
LPS-77699 Update Translations

Commit:
LPS-84119 Auto SF

alecsloan committed:
COMMERCE-1325 Modify Upgrade process to create and transfer catalogs

alecsloan committed:
COMMERCE-1325 Use Catalog groupId when searching for products, so we need to add catalog service to display contexts

Commit:
COMMERCE-1340 Semantic versioning (gw baseline)

alecsloan committed:
COMMERCE-1321 gradle buildLang

alecsloan committed:
COMMERCE-1321 gradle buildService

Commit:
COMMERCE-1340 Source formatting

alecsloan committed:
COMMERCE-1321 gradle baseline

alecsloan committed:
COMMERCE-1321 Move new channel jsps to channel module

alecsloan committed:
COMMERCE-1321 Overview tab isn't needed since there are no other tabs on its level

alecsloan committed:
COMMERCE-1326 Migrate content to catalog scope in upgrade process

Commit:
COMMERCE-1321 Fix Channel typeSetting saving and displaying

alecsloan committed:
COMMERCE-1318 Make sure that user modals have unique ids on the role inputs

alecsloan committed:
COMMERCE-1318 Implement Catalog Details/Users sections SQUASH

alecsloan committed:
COMMERCE-1325 Use Catalog groupId when searching for products, so we need to add catalog service to display contexts SQUASH

Commit:
COMMERCE-1390 Use resourceBundle to properly translate; simplify

Commit:
COMMERCE-1392 Clean up JSPs a little

Commit:
COMMERCE-1392 Build language properties (gw buildLang)

Commit:
COMMERCE-1392 Project dpendencies

Commit:
COMMERCE-1392 Semantic versioning (gw baseline)

alessiorendina committed:
COMMERCE-1408 Fixed tax configuration dto converter

alessiorendina committed:
COMMERCE-X Fixed compile test integration

alecsloan committed:
COMMERCE-1318 Fix orSearch toggle not persisting after jspContributor change

alecsloan committed:
COMMERCE-1321 Localize filter ruleTypes

Commit:
LPS-84119 Fix uses of string.equals

marco-leo committed:
COMMERCE-1340 Add dependencies

marco-leo committed:
COMMERCE-1340 Add provider

marco-leo committed:
COMMERCE-1340 Refactor Taglib

marco-leo committed:
COMMERCE-1340 Use new taglib

marco-leo committed:
COMMERCE-1340 Update taglib version

Commit:
COMMERCE-1321 Properly scope channel-rules to relevant channel

alecsloan committed:
COMMERCE-1321 Implement breadcrumbs for channels

alecsloan committed:
COMMERCE-1316 Implement breadcrumbs for catalogs

alessiorendina committed:
COMMERCE-827 Added missing resource actions

alessiorendina committed:
COMMERCE-827 Added new localized key

alessiorendina committed:
COMMERCE-827 BuildLang

gianmarcobrunialti committed:
COMMERCE-686 add to cart button fixes

gcmznt committed:
COMMERCE-1297 new placeholder images

alessiorendina committed:
COMMERCE-1144 Fixed commerce account helper implementation

gianmarcobrunialti committed:
COMMERCE-686 add to cart button fixes #2

gianmarcobrunialti committed:
COMMERCE-686 add to cart buttons fixes #3

Commit:
COMMERCE-1316 COMMERCE-1321 Add proper description for new portlets

alecsloan committed:
COMMERCE-1319 Implement catalog filter type and jspContributorDisplayContext interface

This is so we can use the same catalog item selector jsp for channel and filter types

alessiorendina committed:
COMMERCE-1409 Fixed account management portlet back url

alessiorendina committed:
COMMERCE-1401 Fixed add to cart tag

alecsloan committed:
COMMERCE-1326 Pass catalogId into product detail

Commit:
LPS-84119 Auto SF

Commit:
LPS-84119 SF didn't feel like fixing this one automatically

alessiorendina committed:
EMP-27 Regenerate package lock

marco-leo committed:
COMMERCE-686 BuildLang

marco-leo committed:
COMMERCE-686 Auto SF

alessiorendina committed:
COMMERCE-1413 Cut old api modules outo of the scope

marco-leo committed:
COMMERCE-1359 Auto SF

marco-leo committed:
COMMERCE-0 Auto SF

Commit:
COMMERCE-X Auto SF

alessiorendina committed:
COMMERCE-1413 Removed source formatter properties and suppressions

alecsloan committed:
LPS-94365 Use latest arquillian version

alessiorendina committed:
COMMERCE-1341 Refactor commerce account display context

alessiorendina committed:
COMMERCE-1341 Refactor commerce account portlet view

alessiorendina committed:
COMMERCE-1341 Added parameter to match constructor

alessiorendina committed:
COMMERCE-1341 Fixed edit commerce account user action command

alessiorendina committed:
COMMERCE-1341 Fixed breccia initializer layouts

marco-leo committed:
COMMERCE-0 Baseline

marco-leo committed:
COMMERCE-0 Baseline

alecsloan committed:
COMMERCE-1316 Implement catalog breadcrumbs in Products' subtabs

liferay-translation committed:
LPS-77699 Update Translations

alecsloan committed:
COMMERCE-1319 Ensure CPRules are scoped so that 'filters' pull from company scope and 'catalog rules' pull from group scope

alecsloan committed:
COMMERCE-1318 Implement default.xml for Catalog resource permissions

alecsloan committed:
COMMERCE-1316 Localize catalog name in breadcrumbs

liferay-translation committed:
LPS-77699 Update Translations

ethanbustad commented:
Merged in https://github.com/liferay/com-liferay-commerce/pull/901.
